list_books.php

```diff
@@ -0,0 +1,34 @@
+<?php
+include 'database.php';
```

Comment:
Observations: The code executes a SQL query without handling potential errors and directly outputs user data without sanitization.\n\nRecommendations: Utilize prepared statements for executing SQL queries and ensure proper error handling. Additionally, sanitize output to prevent XSS attacks. Here's an example using PDO:

```php
$query = "SELECT * FROM books";
$stmt = $connection->prepare($query);
$stmt->execute();
$books = $stmt->fetchAll(PDO::FETCH_ASSOC);

foreach ($books as $row) {
    echo '<tr>';
    echo '<td>' . htmlspecialchars($row['title']) . '</td>';
    echo '<td>' . htmlspecialchars($row['author']) . '</td>';
    echo '<td>' . htmlspecialchars($row['year']) . '</td>';
    echo '<td><a href="edit_book.php?id=' . $row['id'] . '">Edit</a> |
          <a href="delete_book.php?id=' . $row['id'] . '">Delete</a></td>';
    echo '</tr>';
}
```

This approach enhances security by preventing SQL injection and XSS vulnerabilities, and ensures that errors are properly handled.